### PR TITLE
Installed improve coverage 20200316

### PR DESCRIPTION
**Pull request by jkeenan**
The unit tests added to t/Installed.t in this p.r. improve the test coverage for ExtUtils::Installed (as measured on Linux via Devel::Cover) as follows:
```
File                           stmt   bran   cond    sub    pod   time  total
---------------------------- ------ ------ ------ ------ ------ ------ ------
Before:
...lib/ExtUtils/Installed.pm   87.1   68.2   66.6   92.5   88.8   83.3   80.2
After:
...lib/ExtUtils/Installed.pm   89.7   71.1   72.7   96.3   88.8   62.6   83.2
```
Please review and merge to master.

Thank you very much.
Jim Keenan

### CODE REVIEW

**Comment by mohawk2** [comment]
Looks good to me. Is there no CI?

**Comment by jkeenan** [comment]
On 3/16/20 6:44 PM, mohawk2 wrote:
> Looks good to me. Is there no CI?
> 

See:
https://travis-ci.org/github/jkeenan/ExtUtils-Install/jobs/663282410
https://ci.appveyor.com/project/jkeenan/extutils-install

It appears that on Travis AUTHOR_TESTING=1 is set, which causes 
t/pod-coverage.t to be run and FAIL.

I'm not making any changes in POD -- and in my own distros I have no use 
for t/pod.t or t/pod-coverage.t.  And I don't think my p.r.s should be 
judged on the basis of a Travis setting I didn't request.

So, how should I shut Travis up with regard to this?

Thank you very much.
Jim Keenan

**Comment by jkeenan** [comment]
> On 3/16/20 6:44 PM, mohawk2 wrote: Looks good to me. Is there no CI?
> See: https://travis-ci.org/github/jkeenan/ExtUtils-Install/jobs/663282410 https://ci.appveyor.com/project/jkeenan/extutils-install It appears that on Travis AUTHOR_TESTING=1 is set, which causes t/pod-coverage.t to be run and FAIL. I'm not making any changes in POD -- and in my own distros I have no use for t/pod.t or t/pod-coverage.t. And I don't think my p.r.s should be judged on the basis of a Travis setting I didn't request. So, how should I shut Travis up with regard to this? Thank you very much. Jim Keenan

In order not to delay evaluation of the pull request, I have moved the pod tests from t/ to xt/.  Travis now passes:  https://travis-ci.org/github/Perl-Toolchain-Gang/ExtUtils-Install/builds/663323717

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+language: perl
+perl:
+  - "5.30"
+  - "5.28"
+  - "5.26"
+  - "5.24"
+  - "5.22"
+  - "5.20"
+  - "5.18"
+  - "5.16"
```
Comment (oalders) [inline .travis.yml]: Might be worth adding a comment as to why the tests begin at 5.16.  Is there a reason they can't begin with earlier versions?

File: t/Installed.t [hunk 2]
```diff
@@ -224,12 +228,29 @@ my $fake_mod_dir = File::Spec->catdir(cwd(), 'auto', 'FakeMod');
     );
     isa_ok( $realei, 'ExtUtils::Installed' );
     isa_ok( $realei->{Perl}{packlist}, 'ExtUtils::Packlist' );
-    ok( exists $realei->{FakeMod}, 
+    ok( exists $realei->{FakeMod},
         'new() with extra_libs should find modules with .packlists');
-    
+
+    #{ use Data::Dumper; local $realei->{':private:'}{Config};
+    #  warn Dumper($realei); }
```
Comment (oalders) [inline t/Installed.t]: Why include commented code?

**Comment by haarg** [comment]
Moving the tests to xt wouldn't be a problem if they were still getting run.  The tests should be failing on travis, because they are broken.  Just not running them doesn't solve the problem.

I've filed #19 to fix the POD tests.

**Comment by jkeenan** [comment]
On 3/17/20 7:59 AM, Olaf Alders wrote:
> *@oalders* commented on this pull request.
> 
> Moving the Pod tests to xt/ makes sense to me.
> 
> ------------------------------------------------------------------------
> 
> In .travis.yml 
> <https://github.com/Perl-Toolchain-Gang/ExtUtils-Install/pull/18#discussion_r393627021>:
> 
>> @@ -0,0 +1,35 @@
> +language: perl
> +perl:
> +  - "5.30"
> +  - "5.28"
> +  - "5.26"
> +  - "5.24"
> +  - "5.22"
> +  - "5.20"
> +  - "5.18"
> +  - "5.16"
> 
> Might be worth adding a comment as to why the tests begin at 5.16. Is 
> there a reason they can't begin with earlier versions?
> 

The .travis.yml and .appveyor.yml files I added are simply the ones I've 
used recently for my own CPAN distros.  They DWIM for me.  ISTR having 
problems with entries for older perls, but if someone can tell me how 
far back we can go on travis and appveyor, I'll modify them.  I just 
don't want spurious failures.

> ------------------------------------------------------------------------
> 
> In t/Installed.t 
> <https://github.com/Perl-Toolchain-Gang/ExtUtils-Install/pull/18#discussion_r393627078>:
> 
>>          'new() with extra_libs should find modules with .packlists');
> -
> +
> +    #{ use Data::Dumper; local $realei->{':private:'}{Config};
> +    #  warn Dumper($realei); }
> 
> Why include commented code?
> 

That was there in the code when I started.  Since my objective was 
simply to improve coverage, I left it alone.  Whoever is the maintainer 
or CPAN releasor should make the call on that, IMO.

Thanks for taking a look.
Jim Keenan

**Comment by oalders** [comment]
> That was there in the code when I started.  Since my objective was
simply to improve coverage, I left it alone.  Whoever is the maintainer
or CPAN releasor should make the call on that, IMO.

That particular code block looks to be an addition.  Maybe it was copy/pasted from elsewhere?  I see similarly commented code later in the PR.

**Comment by jkeenan** [comment]
> > That was there in the code when I started.  Since my objective was
> > simply to improve coverage, I left it alone.  Whoever is the maintainer
> > or CPAN releasor should make the call on that, IMO.
> 
> That particular code block looks to be an addition. Maybe it was copy/pasted from elsewhere? I see similarly commented code later in the PR.

It's found at line 231 of t/Installed.t in the master branch.

**Pull request closed without merging**